src/CMakeLists.txt

```diff
@@ -79,7 +91,10 @@ endif(CMAKE_CXX_COMPILER_ID STREQUAL GNU)
 set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} ${CMAKE_CXX_FLAGS}")
 
 if(NOT CMAKE_BUILD_TYPE)
-  if(EXISTS "${CMAKE_SOURCE_DIR}/.git")
+  if(EXISTS "${CMAKE_SOURCE_DIR}/.git" AND NOT MINGW)
```

Comment:
instead of encoding the platform specific restrictions, i'd like to provide a machinary which allows us to build ceph by passing different settings.

Reply:
I'm not sure I understand what you mean. We're just setting a default that can still be overridden.

Reply:
i see. what i suggest is to put `-DCMAKE_BUILD_TYPE=RelWithDebInfo" in the build script, instead of putting it here. if we keep on doing so, the cmake script will be very cluttered (even cluttered than what it is now!).

Reply:
Makes sense.